File: examples/node/city-prayer-times.mjs
const city = process.argv[2] ?? "paris";
const language = process.argv[3] ?? "en";
const url = new URL(
  `/api/public/cities/${encodeURIComponent(city)}/prayer-times`,
  "https://pray.zone",
);

url.searchParams.set("lang", language);

const response = await fetch(url, {
  headers: { Accept: "application/json" },
});

if (!response.ok) {
  const message = await response.text();
  throw new Error(`PrayerZone API ${response.status}: ${message}`);
}

const payload = await response.json();
console.log(`${payload.city.name} · ${payload.data.date} · ${payload.data.timezone}`);

for (const prayer of payload.data.prayerTimes) {
  console.log(`${prayer.isNext ? "→" : " "} ${prayer.name.padEnd(10)} ${prayer.time}`);
}

File: examples/python/city_prayer_times.py
import json
import sys
import urllib.parse
import urllib.request

city = sys.argv[1] if len(sys.argv) > 1 else "paris"
language = sys.argv[2] if len(sys.argv) > 2 else "en"
path = f"/api/public/cities/{urllib.parse.quote(city, safe='')}/prayer-times"
query = urllib.parse.urlencode({"lang": language})
request = urllib.request.Request(
    f"https://pray.zone{path}?{query}",
    headers={
        "Accept": "application/json",
        "User-Agent": "PrayerZoneExamples/1.0 (+https://pray.zone)",
    },
)

with urllib.request.urlopen(request, timeout=15) as response:
    payload = json.load(response)

print(f"{payload['city']['name']} · {payload['data']['date']}")
for prayer in payload["data"]["prayerTimes"]:
    marker = "→" if prayer["isNext"] else " "
    print(f"{marker} {prayer['name']:<10} {prayer['time']}")
